visualization/app/codeCharta/ui/customConfigs/customConfigs.component.ts

```diff
@@ -93,9 +103,41 @@ export class CustomConfigsController implements FilesSelectionSubscriber {
 		}, 100)
 	}
 
-	removeCustomConfig(viewId, viewName) {
-		CustomConfigHelper.deleteCustomConfig(viewId)
-		this.dialogService.showInfoDialog(`${viewName} deleted.`)
+	removeCustomConfig(configId, configName) {
+		CustomConfigHelper.deleteCustomConfig(configId)
+		this.dialogService.showInfoDialog(`${configName} deleted.`)
+	}
+
+	downloadPrefetchedCustomConfigs() {
+		if (!this.downloadableConfigs.size) {
+			return
+		}
+
+		CustomConfigHelper.downloadCustomConfigs(this.downloadableConfigs, this.customConfigFileStateConnector)
+	}
+
+	private prefetchDownloadableConfigsForUploadedMaps() {
+		this.downloadableConfigs.clear()
+		const customConfigs = CustomConfigHelper.getCustomConfigs()
+
+		for (const [key, value] of customConfigs.entries()) {
+			// Only Configs which are applicable for at least one of the uploaded maps should be downloaded.
+			if (this.isConfigApplicableForUploadedMaps(value)) {
+				this.downloadableConfigs.set(key, CustomConfigHelper.createExportCustomConfigFromConfig(value))
+			}
+		}
+
+		return (this._viewModel.hasDownloadableConfigs = this.downloadableConfigs.size > 0)
+	}
+
+	private isConfigApplicableForUploadedMaps(customConfig: CustomConfig) {
+		const mapChecksumsOfConfig = customConfig.mapChecksum.split(";")
+		for (const checksumOfConfig of mapChecksumsOfConfig) {
```

Comment:
Yes, I think so. I must admit, I am not a big fan of such methods. I think my way is bit more intuitive especially for new colleagues without deep JS understanding (like me :)). Are there any other advantages of your approach except that it is much shorter?

Reply:
It's definitely not a big thing, so I am fine with keeping it.

For me `.some` is a very little more readable as it uses a builtin function instead of custom logic. It also reduces a very very little the chance to introduce new bugs. It could also be refactored to a one liner, which would reduce the need / huge effort of coming up with local variable names :P I am in general a fan of functional approaches

Reply:
Thanks for the explanation!